opt/scopeserver/__init__.py

```diff
@@ -69,45 +70,20 @@ def run(self) -> None:
 
 
 def log_ascii_header() -> None:
-    logger.info(
-        """\n
-
-  /$$$$$$   /$$$$$$                                       /$$$$$$
- /$$__  $$ /$$__  $$                                     /$$__  $$
-| $$  \__/| $$  \__/  /$$$$$$   /$$$$$$   /$$$$$$       | $$  \__/  /$$$$$$   /$$$$$$  /$$    /$$ /$$$$$$   /$$$$$$
-|  $$$$$$ | $$       /$$__  $$ /$$__  $$ /$$__  $$      |  $$$$$$  /$$__  $$ /$$__  $$|  $$  /$$//$$__  $$ /$$__  $$
- \____  $$| $$      | $$  \ $$| $$  \ $$| $$$$$$$$       \____  $$| $$$$$$$$| $$  \__/ \  $$/$$/| $$$$$$$$| $$  \__/
- /$$  \ $$| $$    $$| $$  | $$| $$  | $$| $$_____/       /$$  \ $$| $$_____/| $$        \  $$$/ | $$_____/| $$
-|  $$$$$$/|  $$$$$$/|  $$$$$$/| $$$$$$$/|  $$$$$$$      |  $$$$$$/|  $$$$$$$| $$         \  $/  |  $$$$$$$| $$
- \______/  \______/  \______/ | $$____/  \_______/       \______/  \_______/|__/          \_/    \_______/|__/
-                              | $$
-                              | $$
-                              |__/
-        """
-    )
+    with open(Path("data") / Path("motd.txt")) as motd:
+        logger.info(motd.read())
 
 
 def generate_config(args) -> Dict[str, Any]:
-    if args.config_file is not None:
-        if not os.path.isfile(args.config_file):
-            raise FileNotFoundError(f"The config file {args.config_file} does not exist!")
+    config = from_file(args.config_file)
 
-        with open(args.config_file, "r") as fh:
-            try:
-                config = json.loads(fh.read())
-            except json.JSONDecodeError:
-                config = {}
-                logger.error("Config file is not proper json and will not be used")
-    else:
-        config = {}
-
-    dhs = config.get("dataHashSecret")
-
-    if not dhs or not dhs.strip():
+    if config.get("dataHashSecret") is None:
```

Comment:
This change allows an empty string to be valid as a secret (which negates its use), the `dhs.strip()` check prevents this.

Reply:
I've added validation in the `config` module for this